include/boost/utility/string_ref.hpp

```diff
@@ -25,7 +26,7 @@
 #include <algorithm>
 #include <iterator>
 #include <string>
-#include <iosfwd>
+#include <ios>
```

Comment:
Is this change needed? `iosfwd` is more lightweight and it should be enough when you're not doing IO. When you do, you've already included `ios` via `ostream` for your stream.

Reply:
For the use of the std::streamsize type (clang/libc++ required it).

Reply:
In fact, `ios` doesn't necessarily forward-declare `basic_ostream`, so the change is not correct.

Reply:
`ios` does declare `std::streamsize`.

Reply:
What if you don't mention `std::streamsize` in the code? Otherwise we'd have to include both headers.

Reply:
Sounds good to me.